code/game/gamemodes/game_mode.dm

```diff
@@ -204,11 +204,10 @@ var/global/list/additional_antag_types = list()
 	for(var/datum/antagonist/antag in antag_templates)
 		antag.post_spawn()
 
-	// Update goals, now that antag status and jobs are both resolved.
+	// Post-initialize, now that antag status and jobs are both resolved.
 	for(var/thing in SSticker.minds)
-		var/datum/mind/mind = thing
-		mind.generate_goals(mind.assigned_job, is_spawning=TRUE)
-		mind.current.show_goals()
+		var/datum/mind/M = thing
+		SEND_SIGNAL(M, COMSIG_MIND_POST_INIT)
```

Comment:
Bruh??? Global signals exist, wtf is this?

Reply:
> Bruh??? Global signals exist, wtf is this?

that signal is also sent on latejoining mobs